### PR TITLE
Use go 1.17 to get clearer information about imported modules

### PR DESCRIPTION
**Pull request by faddat**
Closes: nothing

## Description

This upgrades the go module two the latest format, which makes identifying indirect imports much easier.

We can now see all modules involved with just a glance.

______

For contributor use:

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/main/README.md#merging-a-pr))

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/cosmos/gaia/v5
 
-go 1.16
+go 1.17
```
Comment (alessio) [inline go.mod:3]: There is no need for this bump Jacob. Please let the user build with whichever version of Go > 1.16 they think it's best.

Comment (faddat) [reply go.mod:3]: You're right that there's no explicit need for it. 

The reasoning is that the go1.17 modules are clearer and easier to read.

Comment (faddat) [reply go.mod:3]: maybe we can have the go1.17 formatting and go1.16 compatibility?  

I will check on that.

Comment (alessio) [reply go.mod:3]: The only possibly significant issue is that the SDK v0.42 is not being built and tested with go 1.17

**Review by faddat — COMMENTED** [review]
I made a PR to the SDK too and, I suppose that it could be backported, but I haven't run into any compatibility issues with go 1.17 and have switched to it entirely.

**Pull request closed without merging**